crates/kubelet/src/state/default.rs

```diff
@@ -0,0 +1,407 @@
+//! Default implementation for state machine graph.
+
+use crate::pod::Pod;
+use crate::pod::Phase;
+use crate::state;
+use crate::state::State;
+use crate::state::Transition;
+use log::error;
+use std::sync::Arc;
+
+#[async_trait::async_trait]
+/// Trait for implementing default state machine.
+pub trait DefaultStateProvider: 'static + Sync + Send {
+    /// A new Pod has been created.
+    async fn registered(&self, _pod: &Pod) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    /// Pull images for containers.
+    async fn image_pull(&self, _pod: &Pod) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    /// Image pull has failed several times.
+    async fn image_pull_backoff(&self, _pod: &Pod) -> anyhow::Result<()> {
+        tokio::time::delay_for(std::time::Duration::from_secs(30)).await;
+        Ok(())
+    }
+
+    /// Mount volumes for containers.
+    async fn volume_mount(&self, _pod: &Pod) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    /// Volume mount has failed several times.
+    async fn volume_mount_backoff(&self, _pod: &Pod) -> anyhow::Result<()> {
+        tokio::time::delay_for(std::time::Duration::from_secs(30)).await;
+        Ok(())
+    }
+
+    /// Start containers.
+    async fn starting(&self, _pod: &Pod) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    /// Running state.
+    async fn running(&self, _pod: &Pod) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    /// Handle any errors, on Ok, will transition to Starting again.
+    async fn error(&self, _pod: &Pod) -> anyhow::Result<()> {
+        tokio::time::delay_for(std::time::Duration::from_secs(30)).await;
+        Ok(())
+    }
+}
+
+//
+// * Would be nice to support passing types to the next state (error messages, etc.).
+// * We probably need to explore a more concise way for describing status patches.
+// * Can we offer a macro that doesnt require a trait?
```

Comment:
The more I keep thinking about this, the more I am thinking it would be cleaner. I am not sure if we can enforce macro parameter names, but if we can we could enforce it so people using the shortcut method would do something like:

```rust
graph!(
    ImagePull => logic_stuff,
    Volumes => logic_stuff,
    ...
)
```

So if someone put something like `MyCustomThing =>` we could make it fail. People with more advanced needs can then glue their own together using `state!`